src/test/java/seedu/address/testutil/TypicalRandomProcedureDetails.java

```diff
@@ -0,0 +1,71 @@
+package seedu.address.testutil;
+
+import seedu.address.model.procedure.Completion;
+import seedu.address.model.procedure.Cost;
+import seedu.address.model.procedure.Date;
+import seedu.address.model.procedure.Information;
+import java.util.Random;
+
+/**
+ * This class holds typical details that can make up a Procedure. There are 3 respective details for
+ * Information and Cost. Since this class was designed for the testing of the autosort feature, the Date field
+ * has 3 additional separate components. They are sorted by time periods. For example, the first index contains 3 dates
+ * that are earlier than the dates in the second index. This is true for the second index with respect to the third
+ * index.
+ */
+
+public class TypicalRandomProcedureDetails {
+
+    public static final Information[] EXAMPLE_INFORMATION = {
+            new Information("Apple Inc."), new Information("Google Inc."), new Information("Singtel Inc.")
+    };
+    public static final Cost[] EXAMPLE_COST = {
+            new Cost("10.00"), new Cost("23.23"), new Cost("76.10")
+    };
+    public static final Date[][] EXAMPLE_DATE_SEPARATED_BY_PERIOD = {
+            {
+                new Date("01/01/2011 10:00"), new Date("10/10/2015 13:45"), new Date("02/03/2013 09:00")
+            },
+            {
+                new Date("25/12/2015 16:00"), new Date("01/02/2016 17:12"), new Date("03/11/2018 12:00")
```

Comment:
OHH, thank you very much for doing this up. Would it be good to add in a leap year for the date as well to test this out?

Reply:
That was a good catch. I edited two dates to be leap years just in case. They still pass the tests.